pkg/query-service/app/http_handler.go

```diff
@@ -3241,3 +3258,67 @@ func (aH *APIHandler) liveTailLogs(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+func (aH *APIHandler) queryRangeV4(ctx context.Context, queryRangeParams *v3.QueryRangeParamsV3, w http.ResponseWriter, r *http.Request) {
+
+	var result []*v3.Result
+	var err error
+	var errQuriesByName map[string]string
+	var spanKeys map[string]v3.AttributeKey
+	if queryRangeParams.CompositeQuery.QueryType == v3.QueryTypeBuilder {
+		// check if any enrichment is required for logs if yes then enrich them
+		if logsv3.EnrichmentRequired(queryRangeParams) {
+			// get the fields if any logs query is present
+			var fields map[string]v3.AttributeKey
+			fields, err = aH.getLogFieldsV3(ctx, queryRangeParams)
+			if err != nil {
+				apiErrObj := &model.ApiError{Typ: model.ErrorInternal, Err: err}
+				RespondError(w, apiErrObj, errQuriesByName)
+				return
+			}
+			logsv3.Enrich(queryRangeParams, fields)
+		}
+
+		spanKeys, err = aH.getSpanKeysV3(ctx, queryRangeParams)
+		if err != nil {
+			apiErrObj := &model.ApiError{Typ: model.ErrorInternal, Err: err}
+			RespondError(w, apiErrObj, errQuriesByName)
+			return
+		}
+	}
+
+	result, err, errQuriesByName = aH.querierV2.QueryRange(ctx, queryRangeParams, spanKeys)
```

Comment:
I see these 2 missing:
1. https://github.com/SigNoz/signoz/blob/c28f367f46f92292f85884e5a0bc195882efd8f5/pkg/query-service/app/http_handler.go#L3053
2. https://github.com/SigNoz/signoz/blob/c28f367f46f92292f85884e5a0bc195882efd8f5/pkg/query-service/app/http_handler.go#L3059-L3067

Reply:
I mentioned in the PR description that post-processing logic such as limit is to follow in separate PR. I am aware of the 2nd context finish not being present. I wanted to read about HTTP semantics and handle it better otherwise I will just copy paste the same block.

Reply:
By HTTP semantics I meant something like this https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/206

Reply:
ok